npm/post-install.js

```diff
@@ -1,30 +1,33 @@
-// @ts-check
-import { family, GLIBC, MUSL } from "detect-libc"
-import { exec } from 'child_process'
-import util from 'util'
-
-const execa = util.promisify(exec)
-const platform = process.platform
-const arch = process.arch
-
-let libcFamily
-family().then((fam) => {
-  libcFamily = fam
-})
-
-let libc
-if (platform === "win32") {
-  libc = "-msvc"
-} else {
-  libc = libcFamily === GLIBC ? "-gnu" : libcFamily === MUSL ? "-musl" : ""
-}
-
-const pkg = `@tailcallhq/core-${platform}-${arch}${libc}`
-
-try {
-  // @ts-ignore
-  const { stdout, stderr } = await execa(`npm install ${pkg}@${version} --no-save`)
-  stderr ? console.log(stderr) : console.log(`Successfully installed optional dependency: ${pkg}`, stdout)
-} catch (error) {
-  console.error(`Failed to install optional dependency: ${pkg}`, error.stderr)
-}
+// @ts-check
+import { family, GLIBC, MUSL } from "detect-libc"
+import { exec } from 'child_process'
+import util from 'util'
+
+const execa = util.promisify(exec)
+const platform = process.platform
+const arch = process.arch
+
+let libcFamily
+family().then((fam) => {
+  libcFamily = fam
+})
+
+let libc
+if (platform === "win32") {
+  libc = "-msvc"
+} else {
+  libc = libcFamily === GLIBC ? "-gnu" : libcFamily === MUSL ? "-musl" : ""
+}
+
+const pkg = `@tailcallhq/core-${platform}-${arch}${libc}`
+
+try {
+  // @ts-ignore
+  const { stdout, stderr } = await execa(`npm install ${pkg}@${version} --no-save`)
+  stderr ? console.log(stderr) : console.log(`Successfully installed optional dependency: ${pkg}`, stdout)
+  // Install Scarf SDK as part of the post-install process
+  const { scarfStdout, scarfStderr } = await execa('npm i --save @scarf/scarf')

```

Comment:
why do we need to install the package additionally on the postinstall step? Shouldn't it be already installed by npm as it is specified in the package.json of tailcall package?

Reply:
It will ensure that the package is installed or should we remove it in the package.json file

Reply:
package manager should ensure that package is installed we don't need to to this manually 🙂

Reply:
ohk i will fix it

Reply:
so, I meant that installing package from postinstall script is a manual action that doesn't rely on definition of dependencies in package.json. And if we don't have a specific reason to do so it'd be better to just specify it in package.json and leave the package manager to their work

Reply:
ohk
so can i make a function in gen-root.ts that does this

Reply:
adding it to the `dependencies` list in gen-root should be enough

Reply:
can you check my coomit

Reply:
and go back to the initial question: do we need to change anything in here in post-install.js?